### PR TITLE
Remove API compatibility references to Knative in docs

### PR DESCRIPTION
**Pull request by vtereso**
# Changes
Remove references in the API compatibility docs regarding Knative resources
Remove clusterRole references to Knative resources

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

```
release-note
```

### CODE REVIEW

File: api_compatibility_policy.md [hunk 1]
```diff
@@ -73,10 +61,8 @@ incompatible change is made.
 
 There are two reasons for this:
 
-- `Task` and `TaskRun` are considered upgraded versions of `Build`, meaning that
-  the APIs benefit from a significant amount of user feedback and iteration
-- Going forward users should use `TaskRun` and `Task` instead of `Build` and
-  `BuildTemplate`, those users should not expect the API to be changed on them
+- `Task` and `TaskRun` are considered upgraded versions of [Build](https://github.com/knative/docs/blob/master/docs/build/builds.md#source) and [BuildTemplate](https://github.com/knative/docs/blob/master/docs/build/build-templates.md), meaning that the APIs benefit from a significant amount of user feedback and iteration
```
Comment (houshengbo) [inline api_compatibility_policy.md]: When we refer to the link in the docs, we are recommended to use the link under knative.dev/docs.
For example, Build should refer to https://knative.dev/docs/build/builds/(I am not sure we should use https://knative.dev/docs/build/builds/#source), and BuildTemplate goes to https://knative.dev/docs/build/build-templates/.

Comment (pmorie) [reply api_compatibility_policy.md]: Minor nit, I think it would be good to qualify build and buildtemplate as "Knative Build and BuildTemplate"

**Comment by bobcatfish** [comment]
Good call, thanks for updating this @vtereso !! 🙏 @pmorie has a good point re. saying "Knative Build" vs just "Build" but I'd like to merge this in as is before it gets into a state where it conflicts with master

/lgtm
/approve
/meow space

**Comment by bobcatfish** [comment]
```
I0619 22:31:25.737] 2019/06/19 22:31:25 Published gcr.io/tekton-prow-4/tpipeline-e2e-img/controller-10a3e32792f33651396d02b6855a6e36@sha256:11b2c2f66e29ba4e6cf217ab29e481425666aa75cbbdde2c2decd66a3b95d76a
I0619 22:31:52.361] 2019/06/19 22:31:52 error processing import paths in "config/controller.yaml": unsupported status code 502; body: <!DOCTYPE html>
I0619 22:31:52.362] <html lang=en>
I0619 22:31:52.362]   <meta charset=utf-8>
I0619 22:31:52.362]   <meta name=viewport content="initial-scale=1, minimum-scale=1, width=device-width">
I0619 22:31:52.362]   <title>Error 502 (Server Error)!!1</title>
I0619 22:31:52.362]   <style>
I0619 22:31:52.363]     *{margin:0;padding:0}html,code{font:15px/22px arial,sans-serif}html{background:#fff;color:#222;padding:15px}body{margin:7% auto 0;max-width:390px;min-height:180px;padding:30px 0 15px}* > body{background:url(//www.google.com/images/errors/robot.png) 100% 5px no-repeat;padding-right:205px}p{margin:11px 0 22px;overflow:hidden}ins{color:#777;text-decoration:none}a img{border:0}@media screen and (max-width:772px){body{background:none;margin-top:0;max-width:none;padding-right:0}}#logo{background:url(//www.google.com/images/branding/googlelogo/1x/googlelogo_color_150x54dp.png) no-repeat;margin-left:-5px}@media only screen and (min-resolution:192dpi){#logo{background:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) no-repeat 0% 0%/100% 100%;-moz-border-image:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) 0}}@media only screen and (-webkit-min-device-pixel-ratio:2){#logo{background:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) no-repeat;-webkit-background-size:100% 100%}}#logo{display:inline-block;height:54px;width:150px}
I0619 22:31:52.363]   </style>
I0619 22:31:52.363]   <a href=//www.google.com/><span id=logo aria-label=Google></span></a>
I0619 22:31:52.363]   <p><b>502.</b> <ins>That’s an error.</ins>
I0619 22:31:52.364]   <p>The server encountered a temporary error and could not complete your request.<p>Please try again in 30 seconds.  <ins>That’s all we know.</ins>
I0619 22:31:52.465] ERROR: Build pipeline installation failed
```

waaaaaat

**Comment by bobcatfish** [comment]
/test pull-tekton-pipeline-integration-tests
--

**Pull request merged**